spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaListenerAnnotationBeanPostProcessor.java

```diff
@@ -787,6 +811,36 @@ private void addFormatters(FormatterRegistry registry) {
 		}
 	}
 
+	/**
+	 * Parse a list of partitions into an array. Example: "0-5,10-15".
+	 * @param partsString the list of partions.
+	 * @return the array.
+	 */
+	public static Integer[] parsePartitions(String partsString) {
+		String[] partsStrings = partsString.split(",");
+		if (partsStrings.length == 1) {
+			return new Integer[] { Integer.parseInt(partsStrings[0].trim()) };
```

Comment:
Why do you exit here early?
Doesn't look like you check for the dash in that single value.
How something like this `0-5` is going to work here?